planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/utils/path.cpp

```diff
@@ -0,0 +1,477 @@
+}
+
+ShiftLine get_lane_changing_shift_line(
+  const Pose & lane_changing_start_pose, const Pose & lane_changing_end_pose,
+  const PathWithLaneId & reference_path, const double shift_length)
+{
+  ShiftLine shift_line;
+  shift_line.end_shift_length = shift_length;
+  shift_line.start = lane_changing_start_pose;
+  shift_line.end = lane_changing_end_pose;
+  shift_line.start_idx = autoware::motion_utils::findNearestIndex(
+    reference_path.points, lane_changing_start_pose.position);
+  shift_line.end_idx = autoware::motion_utils::findNearestIndex(
+    reference_path.points, lane_changing_end_pose.position);
+
+  return shift_line;
+}
+
+ShiftedPath get_shifted_path(
+  const PathWithLaneId & target_lane_reference_path, const LaneChangeInfo & lane_change_info)
+{
+  const auto & shift_line = lane_change_info.shift_line;
+  const auto longitudinal_acceleration = lane_change_info.longitudinal_acceleration;
+  const auto lane_change_velocity = lane_change_info.velocity;
+  ShiftedPath shifted_path;
+  PathShifter path_shifter;
+  path_shifter.setPath(target_lane_reference_path);
+  path_shifter.addShiftLine(shift_line);
+  path_shifter.setLongitudinalAcceleration(longitudinal_acceleration.lane_changing);
+
+  // offset front side
+  bool offset_back = false;
+
+  const auto initial_lane_changing_velocity = lane_change_velocity.lane_changing;
+  path_shifter.setVelocity(initial_lane_changing_velocity);
+  path_shifter.setLateralAccelerationLimit(std::abs(lane_change_info.lateral_acceleration));
+
+  if (!path_shifter.generate(&shifted_path, offset_back)) {
+    RCLCPP_DEBUG(
+      autoware::behavior_path_planner::utils::lane_change::get_logger(),
+      "Failed to generate shifted path.");
+  }
+  // TODO(Zulfaqar Azmi): have to think of a more feasible solution for points being remove by
+  // path shifter.
+  return shifted_path;
+}
+
+std::optional<double> exceed_yaw_threshold(
+  const PathWithLaneId & prepare_segment, const PathWithLaneId & lane_changing_segment)
+{
+  const auto & prepare = prepare_segment.points;
+  const auto & lane_changing = lane_changing_segment.points;
+
+  if (prepare.size() <= 1 || lane_changing.size() <= 1) {
+    return std::nullopt;
+  }
+
+  const auto & p1 = std::prev(prepare.end() - 1)->point.pose;
+  const auto & p2 = std::next(lane_changing.begin())->point.pose;
+  const auto yaw_diff = std::abs(autoware::universe_utils::normalizeRadian(
+    tf2::getYaw(p1.orientation) - tf2::getYaw(p2.orientation)));
+  if (yaw_diff > autoware::universe_utils::deg2rad(5.0)) {
+    return yaw_diff;
+  }
+  return std::nullopt;
+}
+
+PathWithLaneId getTargetSegment(
+  const CommonDataPtr & common_data_ptr, const lanelet::ConstLanelets & target_lanes,
+  const Pose & lane_changing_start_pose, const double target_lane_length,
+  const double lane_changing_length, const double lane_changing_velocity,
+  const double buffer_for_next_lane_change)
```

Comment:
I think you can get `target_lanes`, `target_lane_length` and `buffer_for_next_lane_change`from `common_data_ptr`
Just to reduce number of inputs

Reply:
since it is only used in the old `calcTerminalLaneChange`, as discussed yesterday, I remove the function together with the `calcTerminalLaneChange` function.